src/main/java/sorald/cli/Cli.java

```diff
@@ -228,16 +237,42 @@ private SoraldConfig createConfig() {
                 split = ",")
         private List<Checks.CheckType> ruleTypes = new ArrayList<>();
 
+        @CommandLine.Option(
+                names = Constants.ARG_TARGET,
+                description =
+                        "The target of this execution (ex. sorald/92d377). This will be included in the json report.")
+        String target;
+
         @Override
         public Integer call() throws Exception {
             List<? extends JavaFileScanner> checks = inferCheckInstances(ruleTypes);
+
+            var statsCollector = new MinerStatisticsCollector();
+
             if (statsOnGitRepos) {
                 List<String> reposList = Files.readAllLines(this.reposList.toPath());
-                MineSonarWarnings.mineGitRepos(
-                        checks, statsOutputFile.getAbsolutePath(), reposList, tempDir);
+
+                new MineSonarWarnings(statsOutputFile == null ? List.of() : List.of(statsCollector))
+                        .mineGitRepos(
+                                checks, minerOutputFile.getAbsolutePath(), reposList, tempDir);
             } else {
-                MineSonarWarnings.mineLocalProject(checks, originalFilesPath.getAbsolutePath());
+                new MineSonarWarnings(statsOutputFile == null ? List.of() : List.of(statsCollector))
+                        .mineLocalProject(checks, originalFilesPath.getAbsolutePath());
             }
+
+            if (statsOutputFile != null) {
+                Map<String, Object> additionalStatData = new HashMap<>();
+                additionalStatData.put(
+                        StatsMetadataKeys.EXECUTION_INFO,
+                        new ExecutionInfo(
+                                spec.commandLine().getParseResult().originalArgs(),
+                                Constants.SORALD_VERSION,
+                                javaVersion,
+                                target));
```

Comment:
Just a tip, as of Java ... 9 I think, you can do stuff like this with `Map.of`

```suggestion
                Map<String, Object> additionalStatData = Map.of(
                        StatsMetadataKeys.EXECUTION_INFO,
                        new ExecutionInfo(
                                spec.commandLine().getParseResult().originalArgs(),
                                Constants.SORALD_VERSION,
                                javaVersion,
                                target));
```

That's nothing that you need to change, just something that you might not be aware of if you (like most people) have been stuck on Java 8 for an eternity. There's also `List.of` and `Set.of`.